main.go

```diff
@@ -130,12 +130,36 @@ type SystemInfo struct {
 	CPUPhysicalCount int              `json:"cpu_physical_count"`
 }
 
+// SampleSummaryInt helps summarisation of uint64 based fields
+type SampleSummaryInt struct {
+	Maximum uint64  `json:"maximum"`
+	Mean    float64 `json:"mean"`
+	Minimum uint64  `json:"minimum"`
+}
+
+// SampleSummaryFloat helps summarisation of float64 based fields
+type SampleSummaryFloat struct {
+	Maximum float64 `json:"maximum"`
+	Mean    float64 `json:"mean"`
+	Minimum float64 `json:"minimum"`
+}
+
+// SampleSummary helps summarisation the more interesting fields.
+type SampleSummary struct {
+	CPUUser         SampleSummaryFloat `json:"cpu_user"`
+	CPUSystem       SampleSummaryFloat `json:"cpu_system"`
+	CPUIowait       SampleSummaryFloat `json:"cpu_iowait"`
+	RSS             SampleSummaryInt   `json:"rss"`
```

Comment:
Nit: Since RSS is strictly related to the process's memory, I wonder if it is worth renaming this to `RSSMemory` to be consistent with `AvailableMemory`?

Reply:
Later edit: as discussed on Slack, I suppose there might be some more changes here once the negative values issue is fixed so feel free to discard this review comment if it's already being tracked in the follow-up PR.